### PR TITLE
Hide payment details for saved ACH recipients

### PR DESCRIPTION
**Pull request by cjdenio**
Here's what the experience looks like when you send an ACH to an existing recipient:

https://github.com/hackclub/hcb/assets/34525547/b44e9898-8e3d-46cf-b2c0-9507c3f7f32a

### CODE REVIEW

File: app/models/ach_transfer.rb [hunk 1]
```diff
@@ -64,16 +64,19 @@ class AchTransfer < ApplicationRecord
...
   validates :recipient_email, format: { with: URI::MailTo::EMAIL_REGEXP, message: "must be a valid email address" }, allow_nil: true
+  validates :recipient_name, presence: true
   validates_presence_of :recipient_email, on: :create
   validate :scheduled_on_must_be_in_the_future, on: :create
   validate on: :create do
     if amount > event.balance_available_v2_cents
       errors.add(:base, "You don't have enough money to send this transfer! Your balance is #{(event.balance_available_v2_cents / 100).to_money.format}.")
     end
   end
+  validate { errors.add(:base, "Recipient must be in the same org") if payment_recipient && event != payment_recipient.event }
```
Comment (garyhtou) [inline app/models/ach_transfer.rb]: good validation!

File: app/models/payment_recipient.rb [hunk 2]
```diff
@@ -34,4 +34,8 @@ class PaymentRecipient < ApplicationRecord
 
   scope :order_by_last_used, -> { includes(:ach_transfers).order("ach_transfers.created_at DESC") }
 
+  def masked_account_number
+    ("•" * (account_number.length - 4)) + account_number.slice(-4, 4)
```
Comment (garyhtou) [inline app/models/payment_recipient.rb]: See comment about `rjust` here
https://github.com/hackclub/hcb/pull/5898#discussion_r1576850498

**Review by garyhtou — DISMISSED** [review]
i didn't test this myself, but LGTM. my one feedback is to make this card look less like a form when not in editing mode.

image

I'd suggest making it look more like an info card with an edit button.

**Comment by cjdenio** [comment]
Screenshot 2024-04-24 at 1 23 11 PM

I've tweaked the payment details card to look a little less like the form! FWIW it's intentional that it has the same dimensions/sizing as the form— otherwise there's a giant layout shift when you click the edit button

**Pull request merged**